### ISSUE TITLE
Exception in thread "main" java.lang.IllegalAccessError: Update to non-static final field

### ISSUE THREAD

**Issue by twcrone** [newrelic/newrelic-java-agent#620]
Details in https://newrelic.atlassian.net/browse/GTSE-12785

**Comment by lluppani** [comment]
Hi! Can I have more details about this issue? Or access to that Jira ticket :-). Thanks!

**Comment by meiao** [comment]
@lluppani we cannot give access to the Jira ticket since it may contain sensitive information.

This issue has been observed by some customers that use Scala and have recently upgraded to 7.4.3 from 7.2.0 or older releases.

**Comment by lluppani** [comment]
Thats why Im asking. We are facing this issue. Can you provide a workaround?

**Comment by meiao** [comment]
For most customers, downgrading to version 6.5.3 worked.
Version 6.5.3 is not vulnerable to the recent Log4J vulnerabilities.

**Comment by lluppani** [comment]
Yes, we were about to test that. Thanks! How would I know when this issue is fixed in 7.4.x versions?

**Comment by meiao** [comment]
We hope to have it fixed in 7.5.0.
To be notified, you can subscribe to this issue or to the releases.

**Comment by lluppani** [comment]
Great! Thanks!

**Comment by coding-bunny** [comment]
Still not working with the 7.5.0 release:

```
Exception in thread "main" java.lang.IllegalAccessError: Update to static final field akka.pattern.package$.MODULE$ attempted from a different method (<init>) than the initializer method <clinit>
	at akka.pattern.package$.<init>(package.scala:43)
	at akka.pattern.package$.<clinit>(package.scala)
	at akka.event.LoggingBus$class.akka$event$LoggingBus$$addLogger(Logging.scala:179)
	at akka.event.LoggingBus$$anonfun$4$$anonfun$apply$4.apply(Logging.scala:114)
	at akka.event.LoggingBus$$anonfun$4$$anonfun$apply$4.apply(Logging.scala:113)
	at scala.util.Success$$anonfun$map$1.apply(Try.scala:237)
	at scala.util.Try$.apply(Try.scala:192)
	at scala.util.Success.map(Try.scala:237)
	at akka.event.LoggingBus$$anonfun$4.apply(Logging.scala:113)
	at akka.event.LoggingBus$$anonfun$4.apply(Logging.scala:110)
	at scala.collection.TraversableLike$WithFilter$$anonfun$map$2.apply(TraversableLike.scala:683)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike$WithFilter.map(TraversableLike.scala:682)
	at akka.event.LoggingBus$class.startDefaultLoggers(Logging.scala:110)
	at akka.event.EventStream.startDefaultLoggers(EventStream.scala:26)
	at akka.actor.LocalActorRefProvider.init(ActorRefProvider.scala:623)
	at akka.actor.ActorSystemImpl.liftedTree2$1(ActorSystem.scala:620)
	at akka.actor.ActorSystemImpl._start$lzycompute(ActorSystem.scala:617)
	at akka.actor.ActorSystemImpl._start(ActorSystem.scala:617)
	at akka.actor.ActorSystemImpl.start(ActorSystem.scala:634)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:142)
	at akka.actor.ActorSystem$.apply(ActorSystem.scala:126)
	at play.api.libs.concurrent.ActorSystemProvider$.start(Akka.scala:289)
	at play.api.libs.concurrent.ActorSystemProvider.get$lzycompute(Akka.scala:254)
	at play.api.libs.concurrent.ActorSystemProvider.get(Akka.scala:253)
	at play.api.libs.concurrent.ActorSystemProvider.get(Akka.scala:249)
	at com.google.inject.internal.ProviderInternalFactory.provision(ProviderInternalFactory.java:81)
	at com.google.inject.internal.BoundProviderFactory.provision(BoundProviderFactory.java:72)
	at com.google.inject.internal.ProviderInternalFactory.circularGet(ProviderInternalFactory.java:61)
	at com.google.inject.internal.BoundProviderFactory.get(BoundProviderFactory.java:62)
	at com.google.inject.internal.SingleParameterInjector.inject(SingleParameterInjector.java:38)
	at com.google.inject.internal.SingleParameterInjector.getAll(SingleParameterInjector.java:62)
	at com.google.inject.internal.ConstructorInjector.provision(ConstructorInjector.java:104)
	at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:85)
	at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:267)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter$1.call(ProviderToInternalFactoryAdapter.java:46)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1103)
	at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
	at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:145)
	at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:41)
	at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:56)
	at com.google.inject.internal.InjectorImpl$2$1.call(InjectorImpl.java:1016)
	at com.google.inject.internal.InjectorImpl.callInContext(InjectorImpl.java:1092)
	at com.google.inject.internal.InjectorImpl$2.get(InjectorImpl.java:1012)
	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1051)
	at play.api.inject.guice.GuiceInjector.instanceOf(GuiceInjectorBuilder.scala:321)
	at play.api.inject.guice.GuiceInjector.instanceOf(GuiceInjectorBuilder.scala:316)
	at play.api.inject.guice.GuiceApplicationBuilder.build(GuiceApplicationBuilder.scala:98)
	at play.api.inject.guice.GuiceApplicationLoader.load(GuiceApplicationLoader.scala:21)
	at play.core.server.ProdServerStart$.start(ProdServerStart.scala:52)
	at play.core.server.ProdServerStart$.main(ProdServerStart.scala:27)
	at play.core.server.ProdServerStart.main(ProdServerStart.scala)
```

**Comment by kford-newrelic** [comment]
Currently blocked by #438

**Comment by kford-newrelic** [comment]
@coding-bunny thanks for the confirmation - we did a deep-dive on this issue and think we may have a way to resolve it for our next release.

In the meantime, using our `6.5.3` agent appears to be a viable workaround.

**Comment by coding-bunny** [comment]
> @coding-bunny thanks for the confirmation - we did a deep-dive on this issue and think we may have a way to resolve it for our next release.
> 
> In the meantime, using our `6.5.3` agent appears to be a viable workaround.

It doesn't.
This one crashes exactly the same as the 7.5.0

**Comment by kford-newrelic** [comment]
Resolved with #699

**Comment by kford-newrelic** [comment]
Tests for the Akka-HTTP instrumentation failed with this issue under `Java 11` and `Scala 2.12`, so we're reopening this issue for further investigation.

Tests for Akka-HTTP and `Scala 2.13` are successful though, so anyone experiencing this problem might want to try the `7.6.0` agent release in a test environment - we'd love some feedback!

Tests for other Scala frameworks (Play, etc) successful for both `Scala 2.12` and `Scala 2.13` passed successfully.

Won't include this on the `7.6.0` release notes, until we can resolve the known issue.

cc @lluppani @coding-bunny

**Comment by lluppani** [comment]
Yes @kford-newrelic, we can test that in the following days but where can we get the version with the fix? Because I'm not seeing `7.6.0` published yet. Thanks!

**Comment by mattbernst** [comment]
@kford-newrelic I'm still seeing this error in our application built with Akka HTTP 10.1 and Scala 2.12. I just tried it with the newly released 7.6.0 agent running on `OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)` under MacOS. (The same problem happens in prod under Linux.)

```
[error] java.lang.IllegalAccessError: Update to non-static final field akka.http.scaladsl.server.Directives$.DoubleNumber attempted from a different method (akka$http$scaladsl$server$PathMatchers$_setter_$DoubleNumber_$eq) than the initializer method <init> 
```

**Comment by kford-newrelic** [comment]
@mattbernst thanks for this feedback - it's helpful!

**Comment by mattbernst** [comment]
@kford-newrelic Would a test case be helpful? I can reproduce the issue with a fairly small amount of code from one of my company's internal repositories. I think I could make a standalone project that just reproduces this issue so I can share it publicly.

**Comment by kford-newrelic** [comment]
@mattbernst yes, that would be awesome if you could - thank you!

**Comment by mattbernst** [comment]
@kford-newrelic I have created a small test project that reproduces the error here: https://github.com/mattbernst/akka-http-new-relic

**Comment by kford-newrelic** [comment]
Thanks @mattbernst , very cool!

**Comment by meiao** [comment]
Closed by PR #876
